### PR TITLE
storage: handle inactive arrays, which don't have a RAID level

### PR DESCRIPTION
**Pull request by chrisMooreNVIDIA**
In probing arrays, if the array is inactive then udev does not report an MD_LEVEL value for the array.  This was causing a KeyError when probe() tried to add the raidlevel key for the device.  This change checks see if MD_LEVEL exists before it tries to access it.

### CODE REVIEW

**Comment by ogayot** [comment]
Same root cause (with potentially different symptoms):

https://bugs.launchpad.net/ubuntu/+source/subiquity/+bug/1975860

**Comment by ogayot** [comment]
Thanks, LGTM!

I would like somebody else to have a look before merging though.

**Comment by mwhudson** [comment]
This seems like an improvement on the status quo for sure. We'll probably need to fix curtin and subiquity too of course.

More generally I wonder if we should include some kind of status information in the output.

**Comment by dbungert** [comment]
Thanks Chris, this looks good.  I was able to do an install with an incrementally started, inactive, device, without needing further changes to the other parts of the Subiquity family.  The behavior at install time is as if the inactive device doesn't exist, and there doesn't seem to be any trouble with adding a device later.

We don't have the robot running yet on the Probert repo, but it looks like the contributor license agreement hasn't been signed yet - would you do so?
https://ubuntu.com/legal/contributors
Once that's done we can merge this.

**Comment by chrisMooreNVIDIA** [comment]
Sorry for the delay, I had to wait for our legal department to approve my signing of the CLA.  Got the approval, and the form has been submitted.

**Review by dbungert — APPROVED** [review]

**Comment by dbungert** [comment]
@chrisMooreNVIDIA - I have built a beta channel snap that includes this fix.  If you like, it can be tested by booting an install ISO with the kernel command line subiquity-chanel=beta, or drop to a command line in the installer and run `sudo snap refresh --beta subiquity`

**Comment by asizalt** [comment]
Hi.
It happens to me from time to time when installing ubuntu 20.04.6
was this bug fixed/deployed in ubuntu 20.04.6 ?

**Pull request merged**
